contract_comm/currency/currency.go

```diff
@@ -194,14 +224,14 @@ func (cc *CurrencyComparator) getExchangeRate(currency *common.Address) (*exchan
 	return val, nil
 }
 
-func (cc *CurrencyComparator) Cmp(val1 *big.Int, currency1 *common.Address, val2 *big.Int, currency2 *common.Address) int {
+func (cc *CurrencyManager) Cmp(val1 *big.Int, currency1 *common.Address, val2 *big.Int, currency2 *common.Address) int {
 	// Short circuit if the fee currency is the same. nil currency => native currency
 	if (currency1 == nil && currency2 == nil) || (currency1 != nil && currency2 != nil && *currency1 == *currency2) {
 		return val1.Cmp(val2)
 	}
```

Comment:
It seems like this logic is duplicated in `ExchangeRate.CmpValues`. Consider removing the checks here as those will always be called.

Reply:
i though of that; but i left it here because avoid doing the `getExchangeRate` call which might result on an EVM call. Maybe not that important now that they are cached

Reply:
please unresolved if you still think we should remove this